### PR TITLE
feat: edit on GH button

### PR DESCRIPTION
**Pull request by floriandwt**
added:
- button to edit directly a docs page on GitHub

Have changed the table of contents because of that.
Based on https://github.com/inlang/internal/issues/127

### CODE REVIEW

**Comment by samuelstroschein** [comment]
@floriandwt who should review this?

**Comment by floriandwt** [comment]
Based on https://github.com/inlang/internal/issues/127

**Comment by floriandwt** [comment]
@NilsJacobsen you mind quickly looking into it? It's basically just the edit button and the client side documentation file + the tableOfContents (tableOfContents is now dynamically generated based on an object above called fileSources, to filter out the original fileSources for the edit button) that was edited.

File: source-code/website/src/pages/documentation/EditButton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+export function EditButton(props: EditButtonProps) {
+	return (
+		<div class="md:pl-9 w-full pt-8 sm:pt-12 max-sm:pb-8">
+			<a
+				href={props.href}
+				class="text-info/80 hover:text-info/100 text-sm font-semibold"
+				target="_blank"
+			>
+				Edit on GitHub
```
Comment (NilsJacobsen) [inline source-code/website/src/pages/documentation/EditButton.tsx:13]: What about adding an icon as a prefix? I think that makes in more clearly clickable. And directly implies what it is. 
https://icon-sets.iconify.design/material-symbols/edit-outline-rounded/ maybe this one. U should be able to import the material symbols like the other icons.

File: source-code/website/src/pages/documentation/index.page.tsx [hunk 2]
```diff
@@ -7,20 +7,48 @@ import { Callout } from "@src/services/markdown/src/tags/Callout.jsx"
...
+			const markdownHref = props.markdown.frontmatter.href
+
+			for (const section of Object.keys(props.processedTableOfContents)) {
+				const documents = props.processedTableOfContents[section]
+
+				if (documents) {
+					for (const document of documents) {
+						if (document.frontmatter && document.frontmatter.href === markdownHref) {
+							const index = documents.indexOf(document)
+							const fileSource = fileSources[section][index]
```
Comment (NilsJacobsen) [inline source-code/website/src/pages/documentation/index.page.tsx:40]: I get an typeerror here, can you resolve it?

**Review by NilsJacobsen — CHANGES_REQUESTED** [review]
See comments

**Review by NilsJacobsen — APPROVED** [review]
That looks great. Affordance of button is great with the pen. Good job.

**Review by NilsJacobsen — COMMENTED** [review]
Let me check if I can get rid of the lint error

**Comment by NilsJacobsen** [comment]
https://github.com/inlang/inlang/issues/1090#issuecomment-1651150433

Let's merge this shouldn't affect this merge.

**Pull request merged**